sycl/doc/extensions/experimental/sycl_ext_oneapi_bindless_images/sycl_ext_oneapi_bindless_images.asciidoc

```diff
@@ -0,0 +1,1826 @@
+  fp16 = 13,
+  fp32 = 14,
+};
+
+enum class image_type : unsigned int {
+  standard = 0,
+  mipmap = 1,
+  interop = 2,
+};
+
+struct image_descriptor {
+  size_t width;
+  size_t height;
+  size_t depth;
+  image_channel_type channel_type;
+  image_channel_order channel_order;
+  image_type type;
+  unsigned int num_levels;
+
+  image_descriptor(sycl::range<1> dims, sycl::image_channel_order channel_order,
+                   sycl::image_channel_type channel_type, 
+                   image_type type = image_type::standard, 
+                   unsigned int num_levels = 1);
+
+  image_descriptor(sycl::range<2> dims, sycl::image_channel_order channel_order,
+                   sycl::image_channel_type channel_type, 
+                   image_type type = image_type::standard, 
+                   unsigned int num_levels = 1);
+
+  image_descriptor(sycl::range<3> dims, sycl::image_channel_order channel_order,
+                   sycl::image_channel_type channel_type, 
+                   image_type type = image_type::standard, 
+                   unsigned int num_levels = 1);
+};
+
+}
+```
+
+The image descriptor represents the image dimensions, channel type, and channel 
+order. A `type` member is also present to allow for implementation of mipmapped 
+and interop images.
+
+The `image_descriptor` shall be default constructible and follow by-value 
+semantics.
+
+[NOTE]
+====
+Additional future `image_type`s _may_ include "layered" and/or "cubemap".
+====
+
+Note that `image_channel_type` and `image_channel_order` existed in SYCL 1.2.1,
+but were removed in SYCL 2020 in favor of a single, unified enum class.
+We propose separating them again to enable better flexibility
+and to avoid combinatorial complexity.
+
+For the `standard` image type, the value of `num_levels` must be `1`.
+
+Only mipmap image types support more than one level.
+
+The `type` member will inform the implementation of the type of image to 
+create, allocate, or free.
+
+
+=== Allocating image memory
+
+The process of creating an image is two-fold:
+allocate an image's memory, then create an image handle from the allocation.
+Allocation of image memory can be achieved in two ways. 
+
+==== Allocating non-USM image memory 
+
+```cpp
+namespace sycl::ext::oneapi::experimental {
+
+struct image_mem_handle {
+  using raw_handle_type = /* implementation defined */;
+  raw_handle_type raw_handle;
+}
+
+class image_mem {
+
+  image_mem();
+  image_mem(const image_mem &) = delete; // no copy-construct
+
+  image_mem(image_mem &&rhs) noexcept; // move-constructor is allowed
+  image_mem(const sycl::context &syclContext, const image_descriptor &desc);
+
+  ~image_mem();
+
+  image_mem &operator=(image_mem &&); // move-assignment is allowed
+  image_mem &operator=(image_mem &) = delete; // no copy-assignment
+
+  image_mem_handle get_handle() const;
+  image_descriptor get_descriptor() const;
+  sycl::context get_context() const;
+
+  sycl::range<3> get_range() const;
```

Comment:
Should document which dimension is `width`, `height`, `depth`.  Since we are using `range` here, why not use `range` also in `image_descriptor`?